.github/workflows/upgrade_e2e.yaml

```diff
@@ -0,0 +1,130 @@
+# ------------------------------------------------------------
+# Copyright (c) Microsoft Corporation and Dapr Contributors.
+# Licensed under the MIT License.
+# ------------------------------------------------------------
+
+name: E2E - Upgrade path tests - KinD
+
+on:
+  pull_request:
+    branches:
```

Comment:
will add push on master and release branches and probably a schedule for the E2E as a separate PR now that the Github Throttling issue in Mac E2E test is mitigated.